Store/src/credits/credits.cs

```diff
@@ -6,20 +6,71 @@ namespace Store;
 
 public static class Credits
 {
+    // Get player credits safely
     public static int Get(CCSPlayerController player)
     {
-        return Instance.GlobalStorePlayers.Single(p => p.SteamID == player.SteamID).Credits;
+        var storePlayer = Instance.GlobalStorePlayers.FirstOrDefault(p => p.SteamID == player.SteamID);
+        if (storePlayer != null)
+        {
+            return storePlayer.Credits;
+        }
+        else
+        {
+            // Handle case where player is not found
+            // Could log an error or throw a custom exception if necessary
+            return -1;
+        }
     }
+    public static int GetOriginal(CCSPlayerController player)

```

Comment:
Why OriginalCredits?

Reply:
Because Credits = GREATEST(Credits + @SetCredits, 0) allows for direct addition of points in MySQL or added from outside .

Reply:
If we follow the original method, if a player is online, their progress will be reset when they leave